app/api/orders.py

```diff
@@ -68,14 +68,16 @@ def is_payment_valid(order, mode):
         return (order.paid_via == 'paypal') and order.transaction_id
 
 
-def check_billing_info(data):
-    if data.get('amount') and data.get('amount') > 0 and not data.get('is_billing_enabled'):
+def is_billing_info(data, order):
+    if order.event.is_billing_info_mandatory and data.get('amount') and data.get('amount') > 0 and not data.get('is_billing_enabled'):
         raise UnprocessableEntity({'pointer': '/data/attributes/is_billing_enabled'},
-                                  "Billing information is mandatory for paid orders")
-    if data.get('is_billing_enabled') and not (data.get('company') and data.get('address') and data.get('city') and
-                                               data.get('zipcode') and data.get('country')):
-        raise UnprocessableEntity({'pointer': '/data/attributes/is_billing_enabled'},
-                                  "Billing information incomplete")
+                                  "Billing information is mandatory for this order.")
+
+def check_billing_info(data, order):
+	if data.get('is_billing_enabled') and not (data.get('company') and data.get('address')
```

Comment:
You are not checking order.event.is_billing_info_mandatory here

Reply:
Why do I need to check the following in hear when the following will be checked in the `is_billing_info` ?